libs/gl-client/src/lnurl/pay/mod.rs

```diff
@@ -81,7 +131,7 @@ fn extract_description(lnurl_pay_request_response: &PayRequestResponse) -> Resul
         serde_json::from_str(&serialized_metadata.to_owned())
             .map_err(|e| anyhow!("Failed to deserialize metadata: {}", e))?;
 
-    for metadata in deserialized_metadata {
+    for metadata in deserialized_metadata.clone() {
```

Comment:
This seems superfluous given that we clone below anyway.

Reply:
Removed